charts/super-agent/templates/uninstall-job.yaml

```diff
@@ -0,0 +1,49 @@
+{{- if .Values.helm.create -}}
+{{- $installJobName := include "newrelic.common.naming.truncateToDNSWithSuffix" ( dict "name" (include "newrelic.common.naming.fullname" .) "suffix" "uninstall-job" ) -}}
+{{- $configMapName := include "newrelic.common.naming.truncateToDNSWithSuffix" ( dict "name" (include "newrelic.common.naming.fullname" .) "suffix" "manifests" ) -}}
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  annotations:
+    helm.sh/hook: pre-delete
+    helm.sh/hook-delete-policy: before-hook-creation,hook-succeeded
+    helm.sh/hook-weight: "10000"
+  name: {{ $configMapName }}
+data:
+  helm-release.yaml: |
+    {{- include "newrelic-super-agent.helm.release" . | nindent 4 }}
+  helm-repository.yaml: |
+    {{- include "newrelic-super-agent.helm.repository" . | nindent 4 }}
+---
+apiVersion: batch/v1
+kind: Job
+metadata:
+  annotations:
+    helm.sh/hook: pre-delete
+    helm.sh/hook-delete-policy: before-hook-creation,hook-succeeded
+    helm.sh/hook-weight: "10010"
+  name: {{ $installJobName }}
+  namespace: {{ .Release.Namespace }}
+spec:
+  ttlSecondsAfterFinished: 120
+  template:
+    spec:
+      restartPolicy: Never
+      serviceAccountName: helm-controller  # Hardcoded on flux2 chart.
+      containers:
+        - name: delete-crs
+          image: bitnami/kubectl
+          command:
+            - kubectl
+            - delete
```

Comment:
is there a race condition after deleting this? Should we wait  some seconds before cleaning everything else? (maybe a sidecar with a 60s sleep
)

Reply:
Deleting an object in Kuberentes is blocking. It will not be deleted until the helm release is removed.

Well... Actually and being completely specific, is blocking until the object is removed from the API Server but Flux controller adds a finalizer to the object and blocks it until the release is uninstalled.